src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -554,28 +553,23 @@ private async Task<XElement> TryDownloadFileAsync(IFileDownloader fileDownloader
 
                 // This code must always succeed.  If it does not, that means that either the server reported bogus data
                 // to the file-downloader, or the file-downloader is serving us bogus data.  In other event, there is
-                // something wrong with those components, and we should both report the issue to watson, and stop doing
+                // something wrong with those components, and we should both report the issue to Watson, and stop doing
                 // the update.
                 try
                 {
                     using var reader = XmlReader.Create(stream, settings);
 
-                    var result = XElement.Load(reader);
+                    var element = XElement.Load(reader);
                     await LogInfoAsync("Converting data to XElement completed", cancellationToken).ConfigureAwait(false);
-                    return result;
-                }
-                catch (Exception e) when (ReportAndDoNotCatch(e))
-                {
-                    throw ExceptionUtilities.Unreachable();
+                    return (element, delay: default);
                 }
-
-                bool ReportAndDoNotCatch(Exception exception)
+                catch (Exception e) when (_service._reportAndSwallowExceptionUnlessCanceled(e, cancellationToken))
```

Comment:
What's with the _service. thing here having a function we're calling as opposed to just using the existing static methods?

Reply:
(this is not blocking for taking this in to 17.4 of course)

Reply:
same q

Reply:
this was for testing purposes.  we actually validate these codepaths in tests, and it allows the test system to inject a helper that fails-fast as opposed to gracefully falling over (like the main code does).